azureadps-2.0/AzureAD/New-AzureADServiceAppRoleAssignment.md

```diff
@@ -18,21 +18,86 @@ New-AzureADServiceAppRoleAssignment -ObjectId <String> [-InformationAction <Acti
 ```
 
 ## DESCRIPTION
-The New-AzureADServiceAppRoleAssignment cmdlet assigns a service principal to an application role in Azure Active Directory (AD).
+The New-AzureADServiceAppRoleAssignment cmdlet assigns an app role from a resource service principal to a user, a group, or another service principal. App roles assigned to service principals are also known as application permissions.
+
+> [!NOTE]
+> The behavior described here applies when `Connect-AzureAD` was called without any parameters, or using a Microsoft-owned application identity. See [Example 4](#example-4--when-connected-using-a-customer-owned-app-or-service-identity) to learn more about the difference when connected using a customer-owned app registration or service identity.  
 
 ## EXAMPLES
 
-### Example 1
+### Example 1: Assign an app role to another service principal
+
+```powershell
+PS C:\> Connect-AzureAD
+PS C:\> New-AzureADServiceAppRoleAssignment -ObjectId $resource.ObjectId -ResourceId $resource.ObjectId -Id $appRole.Id -PrincipalId $client.ObjectId
+```
+
+In this example, a client service principal is assigned an app role (application permission) defined by a resource service principal (e.g. an API):
+
+- `ObjectId`:  The ObjectId of the resource service principal (e.g. an API).
+- `ResourceId`: The ObjectId of the resource service principal (e.g. an API).
+- `Id`: The Id of the app role (defined on the resource service principal) to assign to the client service principal. If no app roles have been defined on the, you can use `00000000-0000-0000-0000-000000000000`.
```

Comment:
Missing a word. "If no app roles have been defined on the..., you can use..." On the what?